app/models/protocol_subscription.rb

```diff
@@ -18,13 +18,20 @@ class ProtocolSubscription < ApplicationRecord
   after_create :schedule_responses
   after_initialize :initialize_filling_out_for
   after_initialize :initialize_end_date
+  has_many :protocol_transfers
 
   validates_uniqueness_of :filling_out_for_id,
                           scope: %i[person_id state],
                           conditions: -> { where(state: ACTIVE_STATE) },
                           if: ->(sub) { sub.person_id != sub.filling_out_for_id }
   scope :active, (-> { where(state: ACTIVE_STATE) })
 
+  def transfer!(transfer_to)
+    protocol_transfers << ProtocolTransfer.new(from: person, to: transfer_to, protocol_subscription_id: id)
+    self.person = person
+    save!
+  end
```

Comment:
Ik vind eigenlijk dat als (vóór de transfer) het `person_id` van deze protocol_subscription gelijk is aan het `filling_out_for_id`, dat deze functie dan ook het `filling_out_for_id` moet aanpassen. Want anders gaat de protocolsubscriptie een totaal andere betekenis hebben (dan wordt het ineens gezien als een mentor protocolsubscriptie terwijl het dat eerst niet was). 

Ik denk dat we in de praktijk self-protocolsubscripties liever niet willen verplaatsen naar een andere persoon maar je weet maar nooit.

Reply:
nice! thx